### PR TITLE
Fix specs for targeted api collection threshold

### PR DESCRIPTION
**Pull request by Ladas**
Because of a wrong merge, the targeted_api_collection_threshold with
value 0 was not tested, so we were not testing targeted refresh that
fetches the whole collection, instead of by 1 record.

### CODE REVIEW

File: spec/models/manageiq/providers/azure/cloud_manager/targeted_refresher_spec.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
         context "with additional settings #{additional_settings}" do
           before(:each) do
             @refresh_settings = refresh_settings.merge(:allow_targeted_refresh => true)
-            @refresh_settings.merge(additional_settings)
```
Comment (Ladas) [inline spec/models/manageiq/providers/azure/cloud_manager/targeted_refresher_spec.rb]: @bronaghs @djberg96 this was the issue, the settings were not merged, so we were not testing the treshold that switches to collect all API records and filter in ruby vs. collect records by 1 from the API.

**Comment by Ladas** [comment]
@bronaghs we should backport it, so we have it tested in g-release, but not as a blocker for this release

**Comment by djberg96** [comment]
@Ladas You've got a couple failing specs.

**Comment by Ladas** [comment]
@djberg96 uf, finally I made it pass. It was passing locally again. The issue is the resource_group ems_ref has .downcase on it. But then the obfuscation is adding the upcase SUBSCRIPTION_ID, so that was breaking the CI specs.

**Comment by djberg96** [comment]
👍

**Pull request merged**